app/code/Magento/InventorySales/Model/IsProductSalableCondition/IsProductSalableConditionChain.php

```diff
@@ -52,7 +70,7 @@ private function validateConditions(array $conditions)
                 throw new LocalizedException(__('Parameter "object" must be present.'));
             }
 
-            if (empty($condition['sort_order'])) {
+            if (empty($condition['required']) && empty($condition['sort_order'])) {
                 throw new LocalizedException(__('Parameter "sort_order" must be present.'));
```

Comment:
Message should be 

Parameter "sort_order" must be present for urequired conditions.'